synfig-core/src/synfig/loadcanvas.cpp

```diff
@@ -2362,6 +2365,13 @@ CanvasParser::parse_dynamic_list(xmlpp::Element *element,Canvas::Handle canvas)
 			String loop=element->get_attribute("loop")->get_value();
 			weightedaverage_value_node->set_loop(is_true(loop));
 		}
+	} else if (element->get_name()=="average") {
+		average_value_node = new synfig::ValueNode_Average(type);
```

Comment:
Shouldn't it pass `canvas` to constructor as `weighted_average` and `bline` in above lines do?

Reply:
It should! Thanks for the hint!